code/modules/mob/living/carbon/human/human_damage.dm

```diff
@@ -307,24 +310,58 @@ In most cases it makes more sense to use apply_damage() instead! And make sure t
 
 // damage MANY limbs, in random order
 /mob/living/carbon/human/take_overall_damage(var/brute, var/burn, var/sharp = 0, var/edge = 0, var/used_weapon = null)
+    if(status_flags & GODMODE)
+        return    //godmode
+    var/list/obj/limb/parts = get_damageable_limbs(80)
+    var/amount_of_parts = length(parts)
+    for(var/obj/limb/L as anything in parts)
+        L.take_damage(brute / amount_of_parts, burn / amount_of_parts, sharp, edge, used_weapon)
+    updatehealth()
+    UpdateDamageIcon()
+
+// damage MANY LIMBS, in random order
+/mob/living/carbon/human/proc/take_overall_armored_damage(var/damage, var/armour_type = ARMOR_MELEE, var/damage_type = BRUTE, var/limb_damage_chance = 80, var/penetration = 0, var/armour_break_pr_pen = 0, var/armour_break_flat = 0)
 	if(status_flags & GODMODE)
-		return	//godmode
-	var/list/obj/limb/parts = get_damageable_limbs()
-	var/update = 0
-	while(parts.len && (brute>0 || burn>0) )
-		var/obj/limb/picked = pick(parts)
+		return    //godmode
+	var/list/obj/limb/parts = get_damageable_limbs(limb_damage_chance)
+	var/amount_of_parts = length(parts)
+	var/armour_config = GLOB.marine_ranged
+	if(armour_type == ARMOR_MELEE)
+		armour_config = GLOB.marine_melee
+	for(var/obj/limb/L as anything in parts)
+		var/armor = getarmor(L, armour_type)
+		var/modified_damage = armor_damage_reduction(armour_config, damage, armor, penetration, 0, 0)
+		L.take_damage(modified_damage / amount_of_parts)
+	updatehealth()
+	UpdateDamageIcon()
 
-		var/brute_was = picked.brute_dam
-		var/burn_was = picked.burn_dam
 
-		update |= picked.take_damage(brute,burn,sharp,edge,used_weapon)
-		brute	-= (picked.brute_dam - brute_was)
-		burn	-= (picked.burn_dam - burn_was)
+/*
+/mob/living/carbon/human/apply_armoured_damage(var/damage = 0, var/armour_type = ARMOR_MELEE, var/damage_type = BRUTE, var/def_zone = null, var/penetration = 0, var/armour_break_pr_pen = 0, var/armour_break_flat = 0)
+	if(damage <= 0)
+		return ..(damage, armour_type, damage_type, def_zone)
 
-		parts -= picked
-	updatehealth()
-	if(update)	UpdateDamageIcon()
+	var/obj/limb/target_limb = null
+	if(def_zone)
+		target_limb = get_limb(check_zone(def_zone))
+	else
+		target_limb = get_limb(check_zone(rand_zone()))
+	if(isnull(target_limb))
+		return FALSE
 
+	var/armor = getarmor(target_limb, armour_type)
+
+	var/armour_config = GLOB.marine_ranged
+	if(armour_type == ARMOR_MELEE)
+		armour_config = GLOB.marine_melee
+
+	var/modified_damage = armor_damage_reduction(armour_config, damage, armor, penetration, 0, 0)
+	apply_damage(modified_damage, damage_type, target_limb)
+
+	return modified_damage
+
+
```

Comment:
remove commented out code all over the pr once you're done with the final implementation